services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java

```diff
@@ -254,11 +259,13 @@ private void addRealmAccess(RealmModel realm, UserModel user, Map<String, Set<St
         getRealmAdminAccess(realm, realmAdminApp, user, realmAdminAccess);
     }
 
-    private void addMasterRealmAccess(UserModel user, Map<String, Set<String>> realmAdminAccess) {
-        session.realms().getRealmsStream().forEach(realm -> {
-            ClientModel realmAdminApp = realm.getMasterAdminClient();
-            getRealmAdminAccess(realm, realmAdminApp, user, realmAdminAccess);
-        });
+    private void addMasterRealmAccess(UserModel user, String realmName, Map<String, Set<String>> realmAdminAccess) {
+        session.realms().getRealmsStream()
+            .filter(r -> realmName == null || realmName.equals(r.getName()))
+            .forEach(realm -> {
+                ClientModel realmAdminApp = realm.getMasterAdminClient();
+                getRealmAdminAccess(realm, realmAdminApp, user, realmAdminAccess);
+            });
```

Comment:
Why list/filter and not just grab the realm by name? Something like this:

```suggestion
        RealmModel realm = session.realms().getRealmByName(realmName);
        getRealmAdminAccess(realm, realmAdminApp, user, realmAdminAccess);
```